lib/configurators/aws/aws.go

```diff
@@ -742,16 +700,27 @@ func buildPolicyDocument(flags configurators.BootstrapFlags, targetCfg targetCon
 		allActions = append(allActions, docdbActions)
 	}
 
-	dbOption := makeDatabaseActionsBuildOption(flags, targetCfg, boundary)
+	dbOption := makeDatabaseActionsBuildOption(flags)
 	for _, dbActions := range allActions {
 		policyDoc.EnsureStatements(dbActions.buildStatement(dbOption))
 		if dbOption.withAuth {
 			requireSecretsManager = requireSecretsManager || dbActions.requireSecretsManager
-			requireIAMEdit = requireIAMEdit || dbActions.requireIAMEdit
 		}
+
+		// some databases require additional STS actions, e.g. sts:TagSession
+		// for DynamoDB.
+		stsAssumeRoleStatements, err := buildSTSAssumeRoleStatements(targetCfg, dbActions.additionalSTSActions...)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		policyDoc.EnsureStatements(stsAssumeRoleStatements...)
```

Comment:
for `sts:AssumeRole`, you don't need the permissions if the role explicitly trusts the host ARN. Is this also true for `sts:TagSession`? I recall you have to always specify it.

Reply:
Good point.

I tested this: you don't need the `sts:TagSession` permission if the target role trust policy names the trusted principal arn explicitly for the `sts:TagSession` action.

For example, I can omit `sts:AssumeRole` permission in role A, but need the `sts:TagSession` permission in role A, if role B has this trust policy:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::*snip*:role/gavin-tf-permission-tester"
            },
            "Action": "sts:AssumeRole"
        },
        {
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::*snip*:root"
            },
            "Action": "sts:TagSession"
        }
    ]
}
```

and visa-versa. So it works the same way as AssumeRole.

Reply:
TIL. thanks!